cpp/src/Slice/SliceUtil.cpp

```diff
@@ -921,3 +951,15 @@ optional<string> Slice::findMetadata(const string& directive, const map<string,
     auto match = metadata.find(directive);
     return (match != metadata.end() ? make_optional(match->second) : nullopt);
 }
+
+void Slice::mergeMetadataInPlace(StringList& m1, const StringList& m2)
```

Comment:
These two functions are not "merging" the list<string> but concatenating them. See list::merge.
I think a single concatMetadata or appendMetadata function would be sufficient.

Reply:
Fair enough, does `combineMetadata` sit better? I'd like to avoid `appendMetadata`, since even though that's what it does now, it's only because we lack advanced metadata logic. When I add it (which I plan to do somewhat soon), it will do more than just simply appending one list to another. So `appendMetadata` will only be a temporarily correct name.

Reply:
You can fix the function name once you change its implementation to have different semantics.

Reply:
Alright, I renamed it to 'appendMetadata' for the time at hand.